### PR TITLE
[VM]: Support SONiC VM on the server

### PR DESCRIPTION
**Pull request by Pterosaur**
Signed-off-by: Ze Gan <ganze718@gmail.com>

### Description of PR

The VEOS cannot support MACsec feature, so to make SONiC VM as the testbed servers is useful for testing some scenarios, like MACsec. This PR improves some ansible playbooks to start/stop SONiV VM.

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?

Some features aren't supported by the default VM image, VEOS.

#### How did you do it?

Add a switch `vm_type` in some ansible playbooks to choose the VM os to `sonic` or `veos`.

#### How did you verify/test it?

Follow this document: [README.testbed.VsSetup.md](https://github.com/Azure/sonic-mgmt/blob/master/docs/testbed/README.testbed.VsSetup.md)
1. To skip this step: [Download an EOS VM image](https://github.com/Azure/sonic-mgmt/blob/master/docs/testbed/README.testbed.VsSetup.md#download-an-eos-vm-image) and to put a sound SONiC image `sonic-vs.img` in `~/veos-vm/images/`
2. To add an extra parameter `-k sonic` on this the [Setup Arista VMs on the server](https://github.com/Azure/sonic-mgmt/blob/master/docs/testbed/README.testbed.VsSetup.md#setup-arista-vms-on-the-server), like `./testbed-cli.sh -m veos_vtb -n 4 -k sonic start-vms server_1 password.txt`.
3. Check that all VMs are up and running by `ansible -m ping -i veos_vtb server_1 -u admin -k` with password `password`
4. To stop all VMs by `./testbed-cli.sh -m veos_vtb -n 4 -k sonic stop-vms server_1 password.txt`

#### Any platform specific information?

None

#### Supported testbed topology if it's a new test case?

None

### Documentation 

docs/testbed/README.testbed.VsSetup.md

### CODE REVIEW

File: ansible/roles/vm_set/templates/sonic_vm.xml.j2 [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<domain type='kvm' xmlns:qemu='http://libvirt.org/schemas/domain/qemu/1.0'>
+  <name>{{ vm_name }}</name>
+  <memory unit='KiB'>3072000</memory>
+  <currentMemory unit='KiB'>3072000</currentMemory>
+  <vcpu placement='static'>4</vcpu>
```
Comment (lguohan) [inline ansible/roles/vm_set/templates/sonic_vm.xml.j2:5]: do we need 4 cores per sonic vm or we can reduce to 2 cores?

File: ansible/roles/vm_set/templates/sonic_vm.xml.j2 [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<domain type='kvm' xmlns:qemu='http://libvirt.org/schemas/domain/qemu/1.0'>
+  <name>{{ vm_name }}</name>
+  <memory unit='KiB'>3072000</memory>
```
Comment (lguohan) [inline ansible/roles/vm_set/templates/sonic_vm.xml.j2:3]: can we reduce the memory requirement to 2gb?

File: ansible/group_vars/vm_host/main.yml [hunk 3]
```diff
@@ -1,7 +1,9 @@
+support_vm_types: [ "veos", "sonic", "ceos" ]
```
Comment (wangxin) [inline ansible/group_vars/vm_host/main.yml:1]: Is `supported_vm_types` a better name?

File: ansible/roles/vm_set/tasks/start.yml [hunk 4]
```diff
@@ -12,29 +17,55 @@
...
+    - name: Check SONiC image
+      stat: path={{ root_path }}/images/{{ sonic_image_filename }}
+      register: img_stat
 
-- name: Download hdd image
-  get_url: url="{{ vm_images_url }}/{{ hdd_image_filename }}?{{ vmimage_saskey }}" dest="{{ root_path }}/images/{{ hdd_image_filename }}"
-  when: not hdd_stat.stat.exists and not skip_image_downloading
+    - name: Fail if there are no SONiC image and skip image downloading is active
+      fail: msg="Please put {{ sonic_image_filename }} to {{ root_path }}/images"
+      when: not img_stat.stat.exists and skip_image_downloading
```
Comment (wangxin) [inline ansible/roles/vm_set/tasks/start.yml:59]: If we do not support downloading sonic-vs.img for now, maybe it is better to only check if the image file exists and do not check flag `skip_image_downloading`.

**Review by wangxin — COMMENTED** [review]
LGTM except some minor comments.

**Comment by wangxin** [comment]
BTW, can you update the document (https://github.com/Azure/sonic-mgmt/blob/master/docs/testbed/README.testbed.VsSetup.md) as well?

**Review by lguohan — APPROVED** [review]

**Comment by lguohan** [comment]
lgtm

**Review by wangxin — APPROVED** [review]

**Pull request merged**